### PR TITLE
test commit

### PR DESCRIPTION
**Pull request by Dastya**

### CODE REVIEW

**Review by andriiyaroslav — APPROVED** [review]

File: src/test/java/core/basesyntax/db/service/impl/FileCsvReaderServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private static FileCsvReaderService fileCsvReaderService;
+
+    @BeforeClass
+    public static void setUp() {
+        fileCsvReaderService = new FileCsvReaderServiceImpl();
+    }
+
+    @Test
+    public void readFromFile_validFilePath_ok() {
+        List<String> expected = new ArrayList<>(List.of(
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileCsvReaderServiceImplTest.java:23]:
```suggestion
        List<String> expected = List.of(
```
fix everywhere where you use such a code

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileParserServiceImplTest {
+    private static FileParserServiceImpl fileParserService;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:12]:
```suggestion
    private static FileParserService fileParserService;
```

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 20)));
+        List<FruitTransaction> actual = fileParserService.parse(test);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_inputIsEmpty_ok() {
+        List<String> test = new ArrayList<>();
+        List<FruitTransaction> expected = new ArrayList<>();
+        List<FruitTransaction> actual = fileParserService.parse(test);
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:36]:
```suggestion
        List<FruitTransaction> expected = new ArrayList<>();
        List<FruitTransaction> actual = fileParserService.parse(Collections.emptyList());
```

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_letterInUpperCase_ok() {
+        List<String> test = new ArrayList<>(List.of("B,apple,20", "b,BANANA,10"));
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:42]:
```suggestion
        List<String> data = List.of("B,apple,20", "b,BANANA,10"));
```

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        List<FruitTransaction> expected = new ArrayList<>(List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 20),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 10)));
+        List<FruitTransaction> actual = fileParserService.parse(test);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_quantityIsNegative_ok() {
+        List<String> test = new ArrayList<>(List.of("b,banana,-30"));
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:52]:
```suggestion
        List<String> data = new ArrayList<>(List.of("b,banana,-30"));
```

File: src/test/java/core/basesyntax/db/strategy/handler/impl/OperationStrategyImplTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+public class OperationStrategyImplTest {
+    private static OperationStrategy operationStrategy;
+
+    @BeforeClass
+    public static void setUp() {
+        operationStrategy = new OperationStrategyImpl();
+    }
+
+    @Test
+    public void get_supplyOperationTest_ok() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/strategy/handler/impl/OperationStrategyImplTest.java:20]:
```suggestion
    public void get_supplyOperation_ok() {
```
 the same with other methods

File: src/test/java/core/basesyntax/db/strategy/handler/impl/OperationStrategyImplTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void get_returnOperationTest_ok() {
+        Class<ReturnOperationHandler> expected = ReturnOperationHandler.class;
+        Class<? extends OperationHandler> actual = operationStrategy
+                .get(FruitTransaction.Operation.RETURN).getClass();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void get_purchaseOperationTest_ok() {
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/strategy/handler/impl/OperationStrategyImplTest.java:44]: add test case when null is passed

File: src/test/java/core/basesyntax/db/strategy/handler/impl/PurchaseOperationHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.model.FruitTransaction;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class PurchaseOperationHandlerTest {
+    private static PurchaseOperationHandler purchaseOperationHandler;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/strategy/handler/impl/PurchaseOperationHandlerTest.java:13]:
```suggestion
    private static OperationHandler purchaseOperationHandler;
```
the same for other implementations test classes

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    @Test
+    public void parse_inputIsEmpty_ok() {
+        List<FruitTransaction> expected = new ArrayList<>();
+        List<FruitTransaction> actual = fileParserService.parse(Collections.emptyList());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_letterInUpperCase_ok() {
+        List<String> data = new ArrayList<>(List.of("B,apple,20", "b,BANANA,10"));
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:40]:
```suggestion
        List<String> data = List.of("B,apple,20", "b,BANANA,10"));
```
fix everywhere where you use such a code

File: src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+import core.basesyntax.db.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileParserServiceImplTest {
+    private static FileParserServiceImpl fileParserService;
```
Comment (Nick97-git) [inline src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java:13]:
```suggestion
    private static FileParserService fileParserService;
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]
